src/main/java/slimeknights/tconstruct/TConstruct.java

```diff
@@ -69,6 +83,19 @@ public TConstruct() {
     pulseManager.enablePulses();
 
     DistExecutor.runWhenOn(Dist.CLIENT, () -> TinkerBook::initBook);
+
+    if(ModLoadingContext.get().getActiveContainer() instanceof FMLModContainer) {
+      IEventBus eventBus = ((FMLModContainer) ModLoadingContext.get().getActiveContainer()).getEventBus();
+      eventBus.register(CommonBlocks.class);
+      eventBus.register(WorldBlocks.class);
+      eventBus.register(GadgetBlocks.class);
+      eventBus.register(DecorativeBlocks.class);
+      eventBus.register(CommonItems.class);
+      eventBus.register(WorldItems.class);
+      eventBus.register(GadgetItems.class);
+      eventBus.register(ToolItems.class);
+      eventBus.register(FoodItems.class);
+    }
```

Comment:
There are event bus subscriber annotation we can use instead of this if we wanted. Ensures it gets registered as soon as possible.